### PR TITLE
ci: skip sleep when not is_live

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/test_utilities/utils.py [hunk 1]
```diff
@@ -273,12 +275,22 @@ def delete_file_if_exists(file_path: str):
         os.remove(file_path)
 
 
+@contextmanager
+def skip_sleep_in_lro_polling(work_only_if_not_live=True):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/test_utilities/utils.py:279]: need to add some comments to this

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/test_utilities/utils.py [hunk 2]
```diff
@@ -273,12 +275,22 @@ def delete_file_if_exists(file_path: str):
...
+    if work_only_if_not_live and is_live():
+        yield
+        return
+
+    from azure.core.polling.base_polling import LROBasePolling
+    with patch.object(LROBasePolling, "_sleep", return_value=None):
+        yield
+
+
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/tests/test_utilities/utils.py:288]: we could fixture-ize this as well and make it an autouse fixture?

File: sdk/ml/azure-ai-ml/tests/test_utilities/utils.py [hunk 3]
```diff
@@ -273,12 +275,22 @@ def delete_file_if_exists(file_path: str):
...
 
 
+@contextmanager
+def skip_sleep_in_lro_polling(work_only_if_not_live=True):
+    if work_only_if_not_live and is_live():
+        yield
+        return
+
+    from azure.core.polling.base_polling import LROBasePolling
+    with patch.object(LROBasePolling, "_sleep", return_value=None):
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/tests/test_utilities/utils.py:285]: one risk : `_sleep` is a private method, so we have to be wary of its stability. Are there any other public methods we we can use?

File: sdk/ml/azure-ai-ml/tests/test_utilities/utils.py (hunk 2 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/tests/test_utilities/utils.py:288]: that way each test that uses a poller automatically picks up this logic (but can opt out of it)

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/test_utilities/utils.py]: I have moved this logic to recorded tests as it seems ought to be a default behavior for all recorded tests. Also, we can avoid impact those non-recorded tests in this way. 

Do you think we still need to provide a way to skip this mock in recorded method.

File: sdk/ml/azure-ai-ml/tests/test_utilities/utils.py (hunk 3 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/test_utilities/utils.py]: I have mocked RequestsTransport.sleep instead. Let's see if it works well

File: sdk/ml/azure-ai-ml/tests/test_utilities/utils.py (hunk 2 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/test_utilities/utils.py]: Well, it seems that recorded_tests also works for other packages. I'll do a refactor.

File: sdk/ml/azure-ai-ml/tests/conftest.py [hunk 4]
```diff
@@ -562,15 +564,29 @@ def bodiless_matching(test_proxy):
...
+
+
 def pytest_configure(config):
     # register customized pytest markers
     for marker, description in [
         ("e2etest", "marks tests as end to end tests, which involve requests to the server"),
         ("unittest", "marks tests as unit tests, which do not involve requests to the server"),
         ("pipeline_test", "marks tests as pipeline tests, which will create pipeline jobs during testing"),
         ("automl_test", "marks tests as automl tests, which will create automl jobs during testing"),
+        ("core_sdk_test", "marks tests as core sdk tests"),
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/conftest.py:583]: is this change comes from another PR?

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/tests/conftest.py:583]: No, it's in this PR, just register some pytest mark to reduce warnings. Can be removed if you have any concern on this

**Review by needuv — APPROVED** [review]

**Pull request merged**